### PR TITLE
Automation editor tweaks

### PR DESCRIPTION
**Pull request by bramkragten**
## Proposed change

- Attribute triggers/conditions
- Copy automation
- Add input_datetime option for time trigger and condition
- Wait for trigger action

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by zsarnett** [comment]
We should make attribute fields disabled or hidden until an entity is selected

File: src/translations/en.json [hunk 1]
```diff
@@ -998,7 +1003,8 @@
                 },
                 "time": {
                   "label": "Time",
-                  "at": "At"
+                  "at": "At time",
+                  "at_input_datetime": "At the value of an date/time input"
```
Comment (zsarnett) [inline src/translations/en.json:1007]:
```suggestion
                  "at_input_datetime": "At date/time input value"
```

Comment (zsarnett) [reply src/translations/en.json:1007]: At least:
```suggestion
                  "at_input_datetime": "At the value of a date/time input"
```

File: src/panels/config/automation/condition/types/ha-automation-condition-numeric_state.ts [hunk 2]
```diff
@@ -20,15 +19,33 @@ export default class HaNumericStateCondition extends LitElement {
...
+        .name=${"entity_id"}
+        @value-changed=${this._valueChanged}
         .hass=${this.hass}
         allow-custom-entity
       ></ha-entity-picker>
+      <ha-entity-attribute-picker
+        .hass=${this.hass}
+        .entityId=${entity_id}
+        .value=${attribute}
+        .name=${"attribute"}
```
Comment (zsarnett) [inline src/panels/config/automation/condition/types/ha-automation-condition-numeric_state.ts:42]:
```suggestion
        name="attribute"
```

**Comment by zsarnett** [comment]
Sorry I know this a draft. But I was looking over it :)

File: src/panels/config/automation/condition/types/ha-automation-condition-numeric_state.ts (hunk 2 above)
Comment (bramkragten) [reply src/panels/config/automation/condition/types/ha-automation-condition-numeric_state.ts:42]: It should be set as property not as attribute.

File: src/components/entity/ha-entity-attribute-picker.ts [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+      <vaadin-combo-box-light
+        .value=${this._value}
+        .allowCustomValue=${this.allowCustomValue}
+        .renderer=${rowRenderer}
+        @opened-changed=${this._openedChanged}
+        @value-changed=${this._valueChanged}
+      >
+        <paper-input
+          .autofocus=${this.autofocus}
+          .label=${this.label ??
```
Comment (ludeeus) [inline src/components/entity/ha-entity-attribute-picker.ts:88]: Did not know `??` existed, nice one!

File: src/panels/config/automation/condition/types/ha-automation-condition-time.ts [hunk 4]
```diff
@@ -1,44 +1,145 @@
...
+            )}
+            name="before"
+            .value=${before?.startsWith("input_datetime.") ? "" : before}
+            @value-changed=${this._valueChanged}
+          ></paper-input>`}
     `;
   }
 
+  private _handleModeChanged(ev: Event) {
+    const target = ev.target as Radio;
```
[the remark this answers is not in the document]
Comment (bramkragten) [reply src/panels/config/automation/condition/types/ha-automation-condition-time.ts:135]: We actually want target here, as that is the element that is changed (in this case the would be the same).

File: src/panels/config/automation/action/types/ha-automation-action-wait_template.ts [hunk 5]
```diff
@@ -37,9 +38,22 @@ export class HaWaitAction extends LitElement implements ActionElement {
         .value=${timeout}
         @value-changed=${this._valueChanged}
       ></paper-input>
+      <br />
+      <ha-formfield label="Continue on timeout">
+        <ha-switch
```
[the remark this answers is not in the document]
Comment (bramkragten) [reply src/panels/config/automation/action/types/ha-automation-action-wait_template.ts:43]: Why checkbox? Checkbox is usually used when there are more options available?

[the remark this answers is not in the document]
Comment (ludeeus) [reply src/panels/config/automation/action/types/ha-automation-action-wait_template.ts:43]: True, never mind.

**Comment by ludeeus** [comment]
WTH: <https://community.home-assistant.io/t/duplicate-automation-button/219646>

File: src/panels/config/automation/action/types/ha-automation-action-wait_template.ts [hunk 6]
```diff
@@ -37,9 +38,24 @@ export class HaWaitAction extends LitElement implements ActionElement {
         .value=${timeout}
         @value-changed=${this._valueChanged}
       ></paper-input>
+      <br />
+      <ha-formfield
+        label="ui.panel.config.automation.editor.actions.type.wait_template.continue_timeout"
```
Comment (ludeeus) [inline src/panels/config/automation/action/types/ha-automation-action-wait_template.ts:43]:
```suggestion
        .label=${this.hass.localize("ui.panel.config.automation.editor.actions.type.wait_template.continue_timeout")}
```

**Review by ludeeus — APPROVED** [review]

**Pull request merged**